DQM/HcalTasks/interface/ZDCQIE10Task.h

```diff
@@ -0,0 +1,38 @@
+#ifndef ZDCQIE10Task_h
+#define ZDCQIE10Task_h
+
+/*
+ *	file:			ZDCQIE10Task.h
+ *	Author:			Quan Wang
+ *	Description:
+ *		Task for ZDC Read out
+ */
+
+#include "DQM/HcalCommon/interface/ElectronicsMap.h"
+
+class ZDCQIE10Task : public DQMEDAnalyzer
+{
+	public:
+		ZDCQIE10Task(edm::ParameterSet const&);
+		~ZDCQIE10Task() override{}
+
+		void bookHistograms(DQMStore::IBooker&,
+		edm::Run const&, edm::EventSetup const&) override;
+
+	protected:
+		void analyze(edm::Event const&, edm::EventSetup const&) override;
+
+		//	tags
+		edm::InputTag	_tagQIE10;
+		edm::EDGetTokenT<QIE10DigiCollection> _tokQIE10;
+
+		//	hcaldqm::Containers
+		std::map<std::string,MonitorElement*>   _cADC_EChannel;
```

Comment:
hi- i had three comments on this PR
1) your data is much better suited for something like a 
std::array< std::array< std::array<MonitorElement *, 6>, 2 > 4 > _cADC_EChannel 
to avoid the endless conversions to and from strings (per hit!)

2) avoid sprintfs

3) avoid unneeded calls to
ib.setCurrentFolder("Hcal/ZDCQIE10Task/Shape_perChannel");

Reply:
Hi @davidlange6 , just made another commit.
1. This is a really sparse array, so I prefer map.
2. sprintf -> string
3. Done.

Reply:
I second comment (1) -- perhaps @davidlange6 's suggest would result in a sparsely populated array (though it's just pointers, so maybe not a big deal), but it wouldn't be hard to come up with a different solution that avoids performing several string conversions and comparisons for every hit. For example, defining separate maps for each of (EM, HAD, LUM, RPD) * (plus, minus), each a `std::map <int, MonitorElement*>` or even just an array of the desired size, or making an enum like`ZDCSubdet_t = {EMP, HADP, LUMP, RPDP, EMM, HADM, LUMM, RPDM};` and storing things in `std::map<ZDCSubdet_t, std::vector<MonitorElement*> >`.

Reply:
Hi @DryRun @davidlange6 , since nobody likes short string optimization, I switched to detector rawId as the key in the new commit.